website/docs/bestpractices/analytics/spark-best-practices.md

```diff
@@ -0,0 +1,156 @@
+        * Reduce inter-AZ traffic costs
+        * Reduce network latency between executors/Pods
+
+Refer to [Inter AZ Network Optimization](/docs/bestpractices/networking#inter-az-network-optimization) for having pods co-locate on the same AZ.
+
+## Karpenter Recommendations for Scaling Compute nodes
+Here are Karpenter recommendations for Spark workloads. For complete Karpenter configuration details, refer [Karpenter documentation](https://karpenter.sh/docs/).
+
+Consider creating separate NodePools for driver and executor pods.
+
+### Driver Nodepool
+The Spark driver is a single pod and manages the entire lifecycle of the Spark application. Terminating spark driver pod, effectively means terminating the entire spark job.
+    * Configure Driver Nodepool to use on-demand nodes only.
+    * Disable `consolidation` on Driver Nodepool.
+    * Use `node selectors` or `taints/tolerations` for placing driver pods on this designated Driver NodePool.
+
+### Executor Nodepool
+#### Configure Spot instances 
+Consider using spot instances for executors to reduce operational costs.
+When spot instances are interrupted, executors will be terminated and rescheduled on available nodes. For details on interruption behaviour and node termination management, refer to the `Handling Interruptions` section.
+
+#### Instance and Capacity type selection
+
+Using multiple instance types in the node pool enables access to various spot instance pools, increasing capacity availability and optimizing for both price and capacity across the available instance options.
+
+With `Weighted Nodepools`, node selection can be optimized using weighted nodepools arranged in priority order. By assigning different weights to each nodepool, you can establish a selection hierarchy, such as: Spot (highest weight), followed by Graviton, AMD, and Intel (lowest weight).
+
+#### Consolidation Configuration
+While enabling `consolidation` for Spark executor pods can lead to better cluster resource utilization, it's crucial to strike a balance with job performance. Frequent consolidation events can result in slower execution times for Spark jobs, as executors are forced to recompute the shuffle data and RDD blocks.
+
+This impact is particularly noticeable in long-running Spark jobs. To mitigate this, it's essential to carefully tune the consolidation interval.
+
+To save intermediate data computed in Spark Executors, consider one of the following options, which are discussed in `Storage section`:
+    * PVC attach
+    * External Shuffle storage
+    * Executor handling (Related to handling interruptions)
+        * `spark.executor.decommission.enabled=true`
+        * `spark.storage.decommission.enabled=true`
+
+#### Handling interruptions during Karpenter Consolidation/Spot Termination
+
+Perform a controlled decommissioning instead of killing an executor. Leverage `TerminationGracePeriod`, be executor aware and save shuffle data before node being decommissioned.
```

Comment:
```suggestion
Perform controlled decommissioning instead of abruptly killing executors when nodes are scheduled for termination. To achieve this: 

     Configure appropriate TerminationGracePeriod values for Spark workloads
     Implement executor-aware termination handling
     Ensure shuffle data is saved before nodes are decommissioned

```